exporters/otlp/src/otlp_http_client.cc

```diff
@@ -668,8 +668,8 @@ opentelemetry::sdk::common::ExportResult OtlpHttpClient::Export(
   request->ReplaceHeader("Content-Type", content_type);
 
   // Send the request
-  std::unique_ptr<ResponseHandler> handler(new ResponseHandler(options_.console_debug));
-  session->SendRequest(*handler);
+  auto handler = std::make_shared<ResponseHandler>(options_.console_debug);
+  session->SendRequest(handler);
```

Comment:
This is good enough by now. The async version of `OtlpHttpClient` is already finished in #1209 . I will rebase from this PR when it's merged.
There are also some other problems when `OtlpHttpClient` run on multiple threads and they are also fixed in #1209 .